dns/dyndns/src/etc/inc/plugins.inc.d/dyndns/phpDynDNS.inc

```diff
@@ -653,12 +653,25 @@ class updatedns
 
                 /* Get IP for your hostname in Route 53 */
                 if (false !== ($a_result = SearchRecords($records['ResourceRecordSets'], "$hostname"))) {
-                    $OldTTL=$a_result[0][TTL];
-                    $OldIP=$a_result[0][ResourceRecords][0];
+                    /**
+                     * if hostname for ipv4 and ipv6 is the same, a_result contains more than 1 item
+                     * we need to get the item that corresponds to the record type
+                     */
+                    $oldTTLResult = null;
+                    $oldIPResult = null;
+                    foreach ($a_result as $resultItem) {
+                        if ($RecordType === $resultItem['Type']) {
+                            $oldTTLResult = $resultItem[TTL];
+                            $oldIPResult = $resultItem[ResourceRecords][0];
+                        }
+                    }
+
+                    $OldTTL=$oldTTLResult;
+                    $OldIP=$oldIPResult;
                 } else {
                     $OldIP="";
                 }
-
+                
```

Comment:
Can you remove the spurious whitespace addition again?

Reply:
How can I do this after having already pushed the changes?

Reply:
you can update the branch with more commits, github updates the PR automatically :)

1.6